silo/views.py

```diff
@@ -958,12 +966,33 @@ def siloDetail(request, silo_id):
     data = []
     query = makeQueryForHiddenRow(json.loads(silo.rows_to_hide))
 
+    """
+    Note:    There is a chance a service gets stuck in "tasks_running" if a service worker terminates unexpectedly and the task id
+    could not be removed from the task.
+    """
+
+    tasks_running = 0
+    tasks_failed = 0
+
+    for task_status in silo.reads.all().values_list('task_status', flat=True):
```

Comment:
As I said before, there's a better way to do it. You don't need to fetch everything and loop through it because you only wanna know the number of tasks based on their status.
Try something like: 
```python
Read.objects.filter(silo=silo.id,
                    task_status__in=[TASK_CREATED, TRASK_IN_PROGRESS]).count()
```
and for the failed task, you can do the same.
```python
Read.objects.filter(silo=silo.id, task_status=TRASK_FAILED).count()
```